pkg/nsx/services/inventory/build.go

```diff
@@ -0,0 +1,159 @@
+package inventory
+
+import (
+	"context"
+	"crypto/sha1" // #nosec G505: not used for security purposes
+	"errors"
+	"fmt"
+	"sort"
+
+	"github.com/vmware/go-vmware-nsxt/common"
+	"github.com/vmware/go-vmware-nsxt/containerinventory"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+)
+
+func (s *InventoryService) BuildPod(pod *corev1.Pod) (retry bool) {
+	log.Info("Add Pod ", "Pod", pod.Name, "namespace", pod.Namespace)
+	retry = false
+	// Calculate the services related to this Pod from pending_add or inventory store.
+	container_application_ids := []string{}
+	if s.pending_add[string(pod.UID)] != nil {
+		containerApplicationInstance := s.pending_add[string(pod.UID)].(containerinventory.ContainerApplicationInstance)
+		container_application_ids = containerApplicationInstance.ContainerApplicationIds
+	}
+	preContainerApplicationInstance := s.applicationInstanceStore.GetByKey(string(pod.UID))
+	if preContainerApplicationInstance != nil && (len(container_application_ids) == 0) {
+		container_application_ids = preContainerApplicationInstance.(containerinventory.ContainerApplicationInstance).ContainerApplicationIds
+	}
+	namespaceName := pod.Namespace
+	namespace := &corev1.Namespace{}
+	err := s.Client.Get(context.TODO(), types.NamespacedName{Name: namespaceName}, namespace)
+	if err != nil {
+		retry = true
+		log.Error(errors.New("Cannot find namespace for Pod"), "Failed to build Pod", pod)
```

Comment:
use fmt.Error

Reply:
If it's just a error without parameter, it seems no need to use fmt